viewer/packages/cad-parsers/src/cad/filterPrimitivesV9.ts

```diff
@@ -0,0 +1,397 @@
+  computeBoundingBoxFromVertexAttributes
+} from '../utilities/computeBoundingBoxFromAttributes';
+import { filterPrimitivesOutsideClipBox } from './filterPrimitivesCommon';
+import * as THREE from 'three';
+
+export function filterGeometryOutsideClipBox(
+  geometryBuffer: THREE.BufferGeometry,
+  type: RevealGeometryCollectionType,
+  clipBox?: THREE.Box3
+): THREE.BufferGeometry | undefined {
+  if (!clipBox) return geometryBuffer;
+
+  if (type === RevealGeometryCollectionType.InstanceMesh || type === RevealGeometryCollectionType.TriangleMesh) {
+    const boundingBox = geometryBuffer.boundingBox;
+    if (!boundingBox || boundingBox.intersectsBox(clipBox)) {
+      return geometryBuffer;
+    }
+    return undefined;
+  }
+
+  const interleavedAttributes = getAttributes(geometryBuffer, THREE.InterleavedBufferAttribute);
+  let newArray: Uint8Array | undefined;
+
+  switch (type) {
+    case RevealGeometryCollectionType.BoxCollection:
+      newArray = filterBoxCollection(interleavedAttributes, clipBox);
+      break;
+    case RevealGeometryCollectionType.CircleCollection:
+      newArray = filterCircleCollection(interleavedAttributes, clipBox);
+      break;
+    case RevealGeometryCollectionType.ConeCollection:
+      newArray = filterConeCollection(interleavedAttributes, clipBox);
+      break;
+    case RevealGeometryCollectionType.EccentricConeCollection:
+      newArray = filterEccentricConeCollection(interleavedAttributes, clipBox);
+      break;
+    case RevealGeometryCollectionType.EllipsoidSegmentCollection:
+      newArray = filterEllipsoidCollection(interleavedAttributes, clipBox);
+      break;
+    case RevealGeometryCollectionType.GeneralCylinderCollection:
+      newArray = filterGeneralCylinderCollection(interleavedAttributes, clipBox);
+      break;
+    case RevealGeometryCollectionType.GeneralRingCollection:
+      newArray = filterGeneralRingCollection(interleavedAttributes, clipBox);
+      break;
+    case RevealGeometryCollectionType.QuadCollection:
+      newArray = filterQuadCollection(interleavedAttributes, clipBox);
+      break;
+    case RevealGeometryCollectionType.TorusSegmentCollection:
+      newArray = filterTorusCollection(interleavedAttributes, clipBox);
+      break;
+    case RevealGeometryCollectionType.TrapeziumCollection:
+      newArray = filterTrapeziumCollection(interleavedAttributes, clipBox);
+      break;
+    case RevealGeometryCollectionType.NutCollection:
+      newArray = filterNutCollection(interleavedAttributes, clipBox);
+      break;
+    default:
+      assertNever(type);
+  }
+
+  if (newArray.length === 0) {
+    return undefined;
+  }
+
+  return createNewBufferGeometry(newArray, geometryBuffer, interleavedAttributes);
+}
+
+const _views = new Map<number, TypedArrayConstructor>([
+  [1, Uint8Array],
+  [4, Float32Array]
+]);
+
+const quadBoundingBox = new THREE.Box3(new THREE.Vector3(-0.5, -0.5, -1e-4), new THREE.Vector3(0.5, 0.5, -1e-4));
```

Comment:
What is the epsilon? Also, should the max be positive z?
```suggestion
const quadBoundingBox = new THREE.Box3(new THREE.Vector3(-0.5, -0.5, -1e-4), new THREE.Vector3(0.5, 0.5, 1e-4));
```

Reply:
The epsilon is just .. An epsilon. I could put it in a variable called `epsilon`. Would that make it better? :thinking:

Reply:
Edit - created a constant `epsilon`. Don't know what else to do

Reply:
I agree it's not a big change - the change from two negative Z-values to one positive matters a lot though :)

Reply:
What I wondered about was why there is a need to have an epsilon here.